core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -827,26 +829,33 @@ public long getRunAwaySplitCount()
         int count = 0;
         for (RunningSplitInfo splitInfo : runningSplitInfos) {
             Duration duration = Duration.succinctNanos(ticker.read() - splitInfo.getStartTime());
-            if (duration.compareTo(LONG_SPLIT_WARNING_THRESHOLD) > 0) {
+            if (duration.compareTo(stuckSplitsWarningThreshold) > 0) {
                 count++;
             }
         }
         return count;
     }
 
-    private static class RunningSplitInfo
+    public Set<TaskId> getStuckSplitTaskIds(Predicate<RunningSplitInfo> filter)
+    {
+        return runningSplitInfos.stream().filter(filter).map(RunningSplitInfo::getTaskId).collect(toImmutableSet());
+    }
+
+    public static class RunningSplitInfo
             implements Comparable<RunningSplitInfo>
     {
         private final long startTime;
         private final String threadId;
         private final Thread thread;
         private boolean printed;
+        private final PrioritizedSplitRunner split;
```

Comment:
nit: Store the task id and split info directly.

Reply:
Please respond (let me know if you missed it or not going to address)

Reply:
Discussed offline. The split info cannot be cached as it changes as the split execution progresses.